zebra-network/src/isolated.rs

```diff
@@ -54,6 +54,37 @@ pub fn connect_isolated<PeerTransport>(
 ) -> impl Future<Output = Result<BoxService<Request, Response, BoxError>, BoxError>>
 where
     PeerTransport: AsyncRead + AsyncWrite + Unpin + Send + 'static,
+{
+    let nil_inbound_service =
+        tower::service_fn(|_req| async move { Ok::<Response, BoxError>(Response::Nil) });
+
+    connect_isolated_with_inbound(network, data_stream, user_agent, nil_inbound_service)
+}
+
+/// Creates an isolated Zcash peer connection using the provided data stream.
+/// This function is for testing purposes only.
+///
+/// See [`connect_isolated`] for details.
+///
+/// # Additional Inputs
+///
+/// - `inbound_service`: a [`tower::Service`] that answers inbound requests from the connected peer.
+///
+/// # Privacy
+///
+/// This function can make the isolated connection send different responses to peers,
+/// which makes it stand out from other isolated connections from other peers.
+pub fn connect_isolated_with_inbound<PeerTransport, InboundService>(
```

Comment:
Question: Would it make sense to make this function be only available for test builds?

```suggestion
#[cfg(any(test, feature = "proptest-impl"))]
pub fn connect_isolated_with_inbound<PeerTransport, InboundService>(
```

Reply:
It already has this exact restriction when *exported* in `lib.rs`.

This function is called from `connect_isolated` to keep their behaviour consistent. So we still have to *compile* it in all builds, including release builds.